codegen/src/main/java/software/amazon/awssdk/codegen/docs/PaginationDocs.java

```diff
@@ -73,10 +79,52 @@ public String getDocsForSyncResponseClass(ClassName clientInterface) {
                         .toString();
     }
 
+    /**
+     * Constructs additional documentation on the async client operation that is appended to the service documentation.
+     */
+    public String getDocsForAsyncOperation() {
+        return CodeBlock.builder()
+                        .add("<p>This is a variant of {@link #$L($T)} operation. "
+                             + "The return type is a custom publisher that can be subscribed to request a stream of response "
+                             + "pages. SDK will internally handle making service calls for you.\n</p>",
+                             operationModel.getMethodName(), requestType())
+                        .add("<p>When the operation is called, an instance of this class is returned.  At this point, "
+                             + "no service calls are made yet and so there is no guarantee that the request is valid. "
+                             + "The subscribe method should be called as a request to stream data. For more info, "
+                             + "see {@link $T#$L($T)}. If there are errors in your "
```

Comment:
This is a question for both sync and async pagination. How do we handle retryable/transient errors while paginating? Is the iterable/publisher considered terminated at that point and the customer has to make a new request? Is there a way they can get the request object to reconnect (i.e. without getting the last response tokens manually)

Reply:
As discussed offline, added the resume() method in the Response.